src/main/java/uk/gov/hmcts/reform/et/syaapi/service/DocumentGenerationService.java

```diff
@@ -0,0 +1,138 @@
+package uk.gov.hmcts.reform.et.syaapi.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Data;
+import org.apache.logging.log4j.util.Strings;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.et.syaapi.models.TornadoDocument;
+
+import java.util.Locale;
+
+/**
+ * This is a service to generate documents given relevant data and the necessary template.
+ * <p/>
+ * <b><i>Note that Docmosis is an instance of Tornado running in an environment.</i></b>
+ * <p/>
+ * This service relies upon Docmosis as it's engine to generate required documents.
+ * <p/>
+ * This relies upon the following configurations to be set at an environment level:
+ * <ul>
+ *     <li>TORNADO_URL</li>
+ *     <li>TORNADO_ACCESS_KEY</li>
+ * </ul>
+ * </p>
+ * Docmosis is typically installed on all environments within HMCTS.  You can see more information about these
+ * environments at: https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1343291506
+ * </p>
+ * <b>Note:</b><br/>
+ * The templates are stored within the repo: https://github.com/hmcts/rdo-docmosis
+ * <br/>
+ * There is a catch.  This applies to all NON-PRODUCTION environments.  The production environment follows a different
+ * path, whereby the template would need to be uploaded to a sharepoint location which is documented in the page above.
+ */
+@Service
+public class DocumentGenerationService {
+
+    private static final String UNKNOWN_TEMPLATE_ERROR = "Unknown Template: ";
+    private static final String INVALID_OUTPUT_FILE_NAME_ERROR = "Invalid output file name: ";
+
+    private final RestTemplate restTemplate;
+    private final ObjectMapper objectMapper;
+    private final String tornadoUrl;
+    private final String tornadoAccessKey;
+
+    /**
+     * Creates a new instance of {@link DocumentGenerationService} with the specified Tornado details to use its
+     * service in generating the document.
+     *
+     * @param restTemplate     the RestTemplate to use for talking with the Tornado service
+     * @param objectMapper     the {@link ObjectMapper} to generate JSON from
+     * @param tornadoUrl       the Tornado URL endpoint to call to generate the document
+     * @param tornadoAccessKey the access key Tornado will require for authentication
+     */
+    public DocumentGenerationService(RestTemplate restTemplate, ObjectMapper objectMapper,
+                                     @Value("${tornado.url:http://localhost:8090/rs/render}") String tornadoUrl,
```

Comment:
Missing value for tornado.url and tornado.access.key in application.yaml

Reply:
Added the values in application.yaml